### PR TITLE
Add a counter and a GetNextId to XamlContextBase for deterministic cl…

### PR DESCRIPTION
**Pull request by danwalmsley**
…osure naming.

### CODE REVIEW

**Comment by danwalmsley** [comment]
Avalonia side:
https://github.com/AvaloniaUI/Avalonia/pull/4394/files

**Review by kekekeks — APPROVED** [review]

File: src/XamlX/Transform/XamlContextBase.cs [hunk 1]
```diff
@@ -12,6 +12,10 @@ class XamlContextBase
     {
         private readonly Dictionary<Type, object> _items = new Dictionary<Type, object>();
```
Comment (Gillibald) [inline src/XamlX/Transform/XamlContextBase.cs:13]: Maybe I am wrong but isn't the count of items here always equal to your _I'dCount?

File: src/XamlX/Transform/XamlContextBase.cs [hunk 2]
```diff
@@ -12,6 +12,10 @@ class XamlContextBase
...
         private readonly List<IXamlAstNode> _parentNodes = new List<IXamlAstNode>();
+        private int _idCount;
+        
+        public int GetNextUniqueContextId () => ++_idCount;
```
Comment (molnard) [inline src/XamlX/Transform/XamlContextBase.cs:17]: If you need thread safety here consider using https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked.increment?view=netcore-3.1

Comment (kekekeks) [reply src/XamlX/Transform/XamlContextBase.cs:17]: We don't, the underlying MSIL manipulation API isn't thread-safe anyway.

**Pull request merged**